src/CMakeLists.txt

```diff
@@ -2,4 +2,4 @@ get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
 get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
 set(gc_opt_libs ${dialect_libs} ${conversion_libs} MLIROptLib)
 add_llvm_executable(gc-opt gc-opt.cpp)
-target_link_libraries(gc-opt ${gc_opt_libs})
\ No newline at end of file
+target_link_libraries(gc-opt PRIVATE ${gc_opt_libs})
```

Comment:
I have a question: MLIR's own tools also addes

```
llvm_update_compile_flags(mlir-cat)
mlir_check_all_link_libraries(mlir-cat)
```

after `target_link_libraries`. Shall we also add the code?

Reply:
Sure, go for it!

Reply:
OK. I have updated the PR to add `llvm_update_compile_flags` and `mlir_check_all_link_libraries` only.